src/components/modals/login_modal.tsx

```diff
@@ -72,43 +98,43 @@ const LoginModal: React.FC<LoginModalProps> = ({
 		setPassword(value);
 		if (isSubmitted) {
 			const { errors } = validatePassword(value);
-			setValidationErrors(prev => ({
+			setValidationErrors((prev) => ({
 				...prev,
-				password: errors.length > 0 ? errors[0] : undefined
+				password: errors.length > 0 ? errors[0] : undefined,
 			}));
 		}
 	};
 
 	const handleLogin = async () => {
 		setIsSubmitted(true);
 		setError("");
+		setIsLoggingIn(true);
 
 		// Validate email
 		if (!validateEmail(username)) {
-			setValidationErrors(prev => ({
+			setValidationErrors((prev) => ({
 				...prev,
-				email: 'Please enter a valid email address'
+				email: "Please enter a valid email address",
 			}));
+			setIsLoggingIn(false);
 			return;
 		}
 
 		// Validate password
 		const passwordValidation = validatePassword(password);
 		if (!passwordValidation.isValid) {
-			setValidationErrors(prev => ({
+			setValidationErrors((prev) => ({
 				...prev,
-				password: passwordValidation.errors[0]
+				password: passwordValidation.errors[0],
 			}));
+			setIsLoggingIn(false);
 			return;
 		}
 
 		try {
 			const response = await axios.post(
-				"http://localhost:5170/api/auth/login",
-				{
-					username,
-					password,
-				}
+				"http://localhost:5170/api/Auth/login",
```

Comment:
Wrong impementation. That url is for testing locally.

A correct implementation would be something like this:
```suggestion
import API_BASE_URL from "../config";

const apiClient = axios.create({
	baseURL: API_BASE_URL,
	headers: {
		"Content-Type": "application/json",
	},
});

export const fetchAuth = async () => {
	try {
		const response = await apiClient.get("/auth/login");
		return response.data;
	} catch (error) {
		console.error("Error fetching authentication:", error);
		throw error;
	}
};
```

This needs to be added in a the existing file "loginAPI.ts" under /lib/ folder
Then in here you should call the fetchAuth function from loginAPI.ts

Also if your base url in config.ts isn't the azure link, but your localhost, you should change it so it will work on production

Reply:
We just use localhost when testing, but we always change it back before merge